### PR TITLE
cart

### PR DESCRIPTION
**Pull request by chin2yere**
Cart feature

### CODE REVIEW

**Comment by chin2yere** [comment]
I have implemented gitignore, and personalized the user cart feature by implementing a post request upon login that fetches the unique cart based on the user Id.

**Comment by zderick** [comment]
@chin2yere can you add a few things to the gitignore? Take a look at this: https://www.nobledesktop.com/learn/git/ignore-files & https://www.pluralsight.com/guides/how-to-use-gitignore-file

Something like that

File: Petits-debuts-api/Routes/users.js [hunk 1]
```diff
@@ -50,45 +66,42 @@ router.post('/users', async (req, res) => {
...
+
+    const user = await User.findOne({ name: username });
+    console.log(user);
+
+    if (!user) {
+      return res.status(401).json({ error: "Invalid username" });
+    }
+
+    // Compare the password
+    //const isValidPassword = await bcrypt.compare(password, user.password);
```
Comment (zderick) [inline Petits-debuts-api/Routes/users.js]: Can you take this comment out?

File: Petits-debuts-api/server.js [hunk 2]
```diff
@@ -1,104 +1,135 @@
...
+    } else if (!existingCart) {
+      // Create a new cart
+      const newCart = await Cart.create({
+        userId: id,
+        cart: cartPlaceHolder,
+        total: totalPlaceHolder,
+      });
+
+      // Set the user in the session
+      // req.session.user = newUser;
```
Comment (zderick) [inline Petits-debuts-api/server.js]: Remember, we don't want to commit commented code that we're not using

File: Petits-debuts-ui/src/Components/Main/Main.jsx [hunk 3]
```diff
@@ -5,12 +5,14 @@ import { Link } from "react-router-dom";
...
 import ProductGrid from "../ProductGrid/ProductGrid";
+import TopBar from "../TopBar/TopBar";
 
 function Main() {
   const { user, updateUser } = useContext(UserContext);
+  const [cart, updateCart] = useState({});
   const [allProducts, setAllProducts] = useState([]);
   const [selectedCategory, setSelectedCategory] = useState("All categories");
-  const [selectedLocation, setSelectedLocation] = useState("All locations");
+  //const [selectedLocation, setSelectedLocation] = useState("All locations");
```
Comment (zderick) [inline Petits-debuts-ui/src/Components/Main/Main.jsx]: And this

File: Petits-debuts-ui/src/Components/ProductCard/ProductCard.jsx [hunk 4]
```diff
@@ -27,6 +32,9 @@ export default function ProductCard({
           </div>
         </div>
       </div>
+      <div className="row-product-card">
+        {/* <CardButtons id={id} service={service} cart={cart} updateCart={updateCart}/> */}
```
Comment (zderick) [inline Petits-debuts-ui/src/Components/ProductCard/ProductCard.jsx]: and this

File: Petits-debuts-api/Routes/users.js [hunk 5]
```diff
@@ -1,47 +1,63 @@
...
+});
 
 // Route for user registration
-router.post('/users', async (req, res) => {
+router.post("/users", async (req, res) => {
   console.log(req);
-  const { username, email, password, address, businessOwner, location } = req.body;
+  const { username, email, password, address, businessOwner, location } =
+    req.body;
   //console.log('Received registration request:', { name, email, password, address, businessOwner, location });
```
Comment (zderick) [inline Petits-debuts-api/Routes/users.js]: Remember we don't want to leave commented code into the commit please

File: Petits-debuts-ui/src/Components/Main/Main.jsx [hunk 6]
```diff
@@ -26,8 +28,43 @@ function Main() {
...
+          body: JSON.stringify({
+            id,
+          }),
+          credentials: "include",
+        });
+
+        if (response.ok) {
+          const data = await response.json();
+          const loggedInUserCart = data.usercart;
+          // console.log(data.usercart);
```
Comment (zderick) [inline Petits-debuts-ui/src/Components/Main/Main.jsx]: Remember we don't want to leave commented code in the commits

File: Petits-debuts-ui/src/Components/ProductGrid/ProductGrid.jsx [hunk 7]
```diff
@@ -15,6 +15,8 @@ export default function ProductGrid({
   selectedCategory,
   setSelectedCategory,
   filterProductsByCategory,
+  cart,
+  updateCart,
 }) {
   //const [url, setUrl] = useState("");
```
Comment (zderick) [inline Petits-debuts-ui/src/Components/ProductGrid/ProductGrid.jsx]: Here

File: Petits-debuts-ui/src/Components/TopBar/TopBar.css [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  margin-right: 10px;
+}
+.button-left {
+  width: 80%;
+  background-color: transparent;
+  outline: none;
+}
+.row-drop-down {
+  display: flex;
+  /* background-color: white; */
```
Comment (zderick) [inline Petits-debuts-ui/src/Components/TopBar/TopBar.css]: Here

**Pull request merged**